pkgs/applications/networking/p2p/synapse-bt/default.nix

```diff
@@ -13,10 +13,13 @@ rustPlatform.buildRustPackage rec {
 
   cargoSha256 = "1sc8c0w2dbvcdv16idw02y35x0jx5ff6ddzij09pmqjx55zgsjf7";
 
-  buildInputs = [ pkgconfig openssl ] ++ stdenv.lib.optional stdenv.isDarwin Security;
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ openssl ] ++ stdenv.lib.optional stdenv.isDarwin Security;
 
   cargoBuildFlags = [ "--all" ];
 
+  PKG_CONFIG_ALLOW_CROSS = 1;
```

Comment:
Is this something we should set by default in `buildRustPackage`?

Reply:
possibly, it's for `pkg-config` so ostensibly it would need to be set anywhere `pkg-config` is used during cross-compilation. Don't know how to handle all those cases though.

Reply:
Actually it is a pkgconfig-rs thing i think

Reply:
Maybe some hook in buildRustPackage that checks if pkg-config is available and sets `PKG_CONFIG_ALLOW_CROSS` in that case?